asynction/types.py

```diff
@@ -178,6 +363,7 @@ class Server:
 
     url: str
     protocol: ServerProtocol
+    security: Optional[Sequence[SecurityRequirement]] = None
```

Comment:
```suggestion
    security: Optional[Sequence[Mapping[str, Sequence[str]] = None
```

Let's keep the types mirroring the actual YAML data.

I do understand that the `SecurityRequirement` class performs some validation, however if we want to properly validate security, we can only do that on the `AsyncAPISpec` object level. This way we can validate scheme name and scopes using the top-lvl `components` field. See comment below.